oap-server/server-configuration/configuration-etcd/src/main/java/org/apache/skywalking/oap/server/configuration/etcd/EtcdConfigWatcherRegister.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.apache.skywalking.oap.server.configuration.etcd;
+
+import java.io.IOException;
+import java.net.URI;
+import java.util.Set;
+import mousio.etcd4j.EtcdClient;
+import mousio.etcd4j.promises.EtcdResponsePromise;
+import mousio.etcd4j.responses.EtcdErrorCode;
+import mousio.etcd4j.responses.EtcdException;
+import mousio.etcd4j.responses.EtcdKeysResponse;
+import org.apache.skywalking.oap.server.configuration.api.ConfigTable;
+import org.apache.skywalking.oap.server.configuration.api.ConfigWatcherRegister;
+import org.apache.skywalking.oap.server.library.module.ModuleStartException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * @author Alan Lau
+ */
+public class EtcdConfigWatcherRegister extends ConfigWatcherRegister {
+
+    private final static Logger logger = LoggerFactory.getLogger(EtcdConfigWatcherRegister.class);
+
+    private EtcdServerSettings settings;
+
+    private EtcdClient client;
+
+    @Override public ConfigTable readConfig(Set<String> keys) {
+
+        if (client == null) {
+            try {
+                client = new EtcdClient(EtcdUtils.parse(settings).toArray(new URI[] {}));
+            } catch (ModuleStartException e) {
+                logger.error(e.getMessage(), e);
+            }
+
+            EtcdResponsePromise<EtcdKeysResponse> promise;
+            try {
+                promise = client.get(settings.getClusterName()).waitForChange().send();
+                promise.addListener(responsePromise -> {
+                    onDataValueChanged();
+                });
```

Comment:
This looks like asking for change every time, is this a good way in etcd? Or is there a change to register and listening for change notification always.

Reply:
Yes, it is a problem for everytime make a waitForChange. I will fix this.

Reply:
As we have executor to sync config every 60 sec, so i need not add a monitor for value changed. I just remove the waitForChange logic. what's your idea ? @wu-sheng 
Another way i add a map to store the waitForChange for every key, everytime i check the key is exists, remove which when the change occurs.  but i think is not needed.  also the logic is to complex. I also can use the timeout, but here, I think it is needed.

Reply:
I am not an etcd user, I can't tell much detail. My concern and question is only, do query for etcd every 60 sec, is it a high payload when oap is a 20-30 nodes cluster.

Reply:
In zk implementation, I think it set up a listener. Yes? @kezhenxu94 
Anyway, you could make the decision, and we could wait for user feedback.

Reply:
Where is the timeout for etcd server access?
Current codes, with more configuration items, etcd server will pay more payload to response all items every time. 
Take a look at @kezhenxu94 's pr about set alarm rule dynamicly. That PR is ready to be merged, but it tells you, etcd server needs to response while alarm rule everytime. I am not sure it is the right way.

Reply:
> In zk implementation, I think it set up a listener. Yes? @kezhenxu94
> Anyway, you could make the decision, and we could wait for user feedback.

It uses `PathChildrenCache`, whose JavaDoc says that it **watch the ZK path**, so technically **YES**, so do the Apollo and Nacos implementations, we use watcher/listener whenever possible.

> My concern and question is only, do query for etcd every 60 sec, is it a high payload when oap is a 20-30 nodes cluster

That's really a concern, even when there're just one or two oap nodes, but the configuration items/contents are large (although we have only several config items, but I believe it definitely grows in the future). So @wayilau please, take this into consideration, and yet, you will make the decision as @wu-sheng said

Reply:
ok, i will find a better way to use waitForChange to achieve this.